workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/NetworkPoliciesValidator.java

```diff
@@ -0,0 +1,99 @@
+import io.hyscale.controller.activity.ValidatorActivity;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.controller.provider.PortsProvider;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.NetworkTrafficRule;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.List;
+
+@Component
+public class NetworkPoliciesValidator implements Validator<WorkflowContext> {
+
+    private static final Logger logger = LoggerFactory.getLogger(NetworkPoliciesValidator.class);
+
+    @Autowired
+    private StructuredOutputHandler structuredOutputHandler;
+
+    @Autowired
+    private PortsProvider portsProvider;
+
+    @Override
+    public boolean validate(WorkflowContext workflowContext) throws HyscaleException {
+        logger.info("Validating Network Policies");
+        ServiceSpec serviceSpec = workflowContext.getServiceSpec();
+        boolean external = false;
+        if (serviceSpec.get(HyscaleSpecFields.external, boolean.class) != null) {
+            external = serviceSpec.get(HyscaleSpecFields.external, boolean.class);
+        }
+        List<NetworkTrafficRule> networkTrafficRules = serviceSpec.get(HyscaleSpecFields.allowTraffic, new TypeReference<>() {
+        });
+
+        if (external && networkTrafficRules != null) {
+            logger.info("External Cannot be True to Apply Network Traffic Rules");
+            addErrorMessage(ValidatorActivity.INVALID_VALUE, HyscaleSpecFields.external);
+            WorkflowLogger.persist(ValidatorActivity.INVALID_VALUE, LoggerTags.ERROR, HyscaleSpecFields.external);
+            return false;
+        }
+        if (external || CollectionUtils.isEmpty(networkTrafficRules)) {
+            return true;
+        }
+        return validateTrafficRules(networkTrafficRules, portsProvider.getExposedPorts(serviceSpec, true));
+    }
+
+    //Check for Valid Network Traffic Rules
+    private boolean validateTrafficRules(List<NetworkTrafficRule> networkTrafficRules, List<Integer> exposedPorts) {
+        for (NetworkTrafficRule networkTrafficRule : networkTrafficRules) {
+            if (networkTrafficRule.getPorts() == null) {
+                logger.info("Network traffic Rules are not Valid");
+                addErrorMessage(ValidatorActivity.INVALID_NETWORK_TRAFFIC_RULES, (String) null);
+                WorkflowLogger.persist(ValidatorActivity.INVALID_NETWORK_TRAFFIC_RULES, LoggerTags.ERROR);
+                return false;
+            }
+            // Rules is invalid if port is not exposed
+            for (Integer port : networkTrafficRule.getPorts()) {
+                if (CollectionUtils.isNotEmpty(exposedPorts) && !exposedPorts.contains(port)) {
```

Comment:
If the service doesn't have any port and Network Traffic rules define a port how will this act?
Also We should specify all the ports that are not exposed, not just the first one.